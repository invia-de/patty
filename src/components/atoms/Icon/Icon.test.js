import React from 'react';
import AllIcons, { Icon } from './Icon';
import { render } from 'react-testing-library';

Object.keys(AllIcons).forEach(manner => {
  if (manner !== 'Icon') {
    const Icon = AllIcons[manner];

    test(manner + ' renders', () => {
      const { container } = render(<Icon />);
      expect(container.firstChild).toMatchSnapshot();
    });
  }
});

test('Icon renders (icon path)', () => {
  const { container } = render(
    <Icon
      path={[
        'M11.917 13.167l-11.917-12.083h32l-12 12.083v15.667l-8.083 2.083z'
      ]}
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
