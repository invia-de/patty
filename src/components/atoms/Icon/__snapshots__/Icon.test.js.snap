// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders (icon path) 1`] = `
<svg
  aria-hidden="true"
  class="icon"
  viewBox="0 0 512 512"
>
  <path
    d="M11.917 13.167l-11.917-12.083h32l-12 12.083v15.667l-8.083 2.083z"
  />
</svg>
`;

exports[`IconArrowLeft renders 1`] = `
<svg
  aria-hidden="true"
  class="icon"
  viewBox="0 0 512 512"
>
  <path
    d="M425 430L249 255 426 80 346 0 87 259l253 253 85-82z"
  />
</svg>
`;

exports[`IconArrowRight renders 1`] = `
<svg
  aria-hidden="true"
  class="icon"
  viewBox="0 0 512 512"
>
  <path
    d="M88 82l176 176L87 432l80 80 259-259L173 0 88 82z"
  />
</svg>
`;
